test/ref_ops_test.cpp

```diff
@@ -7188,6 +7188,67 @@ TEST_CASE(slice_test)
     }
 }
 
+TEST_CASE(slice_dyn_test0)
+{
+    // Slice a single dynamic dimension
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape s{migraphx::shape::int32_type, {{2, 3, 0}, {2, 2, 0}, {3, 3, 0}}};
+    auto x = mm->add_parameter("x", s);
+    mm->add_instruction(migraphx::make_op("slice", {{"axes", {2}}, {"starts", {1}}, {"ends", {3}}}),
+                        x);
+    migraphx::shape s2{migraphx::shape::int32_type, {{2, 3, 0}, {2, 2, 0}, {2, 2, 0}}};
+    EXPECT(p.get_output_shapes().back() == s2);
+    p.compile(migraphx::ref::target{});
+
+    //  the strides of sresult are those of the original shape, not
+    // reduced to sliced size.
+    migraphx::shape sresult{migraphx::shape::int32_type, {2, 2, 2}, {6, 3, 1}};
+    migraphx::shape input_fixed_shape{migraphx::shape::int32_type, {2, 2, 3}};
+    migraphx::parameter_map params;
+    std::vector<int> data(2 * 2 * 3);
+    std::iota(data.begin(), data.end(), 0);
+    params["x"] = migraphx::argument(input_fixed_shape, data.data());
+    auto result = p.eval(params).back();
+
+    std::vector<int> gold = {1, 2, 4, 5, 7, 8, 10, 11};
+    std::vector<int> results_vector(2 * 2 * 2);
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    EXPECT(migraphx::verify_range(results_vector, gold));
+    EXPECT(result.get_shape() == sresult);
+}
+
+TEST_CASE(slice_dyn_test1)
+{
+    // Slice all three dynamic dimensions
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape s{migraphx::shape::int32_type, {{2, 2, 0}, {2, 2, 0}, {3, 3, 0}}};
```

Comment:
Can you make this not fixed lengths ? 
also between these two `slice_dyn_test0` and `slice_dyn_test1` make one of the test such where slice_lengths is greater than`max` allowed length and in another axis or test  slice length is lower than `min` length.

Reply:
Done.  Note that for the min case the result is perfectly sensible:  the sliced tensor is smaller than the original and the output shape is a smaller dimension than the input shape.